src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractLanguageServer.cs

```diff
@@ -211,61 +201,40 @@ public DelegatingEntryPoint(string method, AbstractLanguageServer<TRequestContex
             });
         }
 
-        /// <summary>
-        /// StreamJsonRpc entry point for all handler methods.
-        /// The optional parameters allow StreamJsonRpc to call into the same method for any kind of request / notification (with any number of params or response).
-        /// </summary>
-        public async Task<JToken?> ExecuteRequestAsync(JToken? request = null, CancellationToken cancellationToken = default)
-        {
-            var queue = _target.GetRequestExecutionQueue();
-            var lspServices = _target.GetLspServices();
-
-            // Retrieve the language of the request so we know how to deserialize it.
-            var language = _target.GetLanguageForRequest(_method, request);
+        public abstract MethodInfo GetEntryPoint(bool hasParameter);
```

Comment:
I created two subclasses of AbstractLanguageServer, one for Newtonsoft and one for STJ (we still support webtools/razor using clasp using newtonsoft).

This can all be simplified once we move Razor/XAML over to our protocol types and no longer need to manually handle serialization